dexterion.cpp

```diff
@@ -28,11 +31,55 @@ LRESULT Wndproc(HWND hWnd, UINT msg, WPARAM wParam, LPARAM lParam) {
 	return DefWindowProc(hWnd, msg, wParam, lParam);
 }
 
+// Function to get the PID of a process by its name
+DWORD GetPIDByName(const wchar_t* processName) {
+	DWORD pid = 0;
+	HANDLE hSnap = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
+	if (hSnap != INVALID_HANDLE_VALUE) {
+		PROCESSENTRY32W pe32; // Using PROCESSENTRY32W for wide strings (wchar_t)
+		pe32.dwSize = sizeof(PROCESSENTRY32W);
+		if (Process32FirstW(hSnap, &pe32)) { // Using Process32FirstW for wide strings (wchar_t)
+			do {
+				if (_wcsicmp(pe32.szExeFile, processName) == 0) { // Using _wcsicmp for wide strings (wchar_t) comparison
+					pid = pe32.th32ProcessID;
+					break;
+				}
+			} while (Process32NextW(hSnap, &pe32)); // Using Process32NextW for wide strings (wchar_t)
+		}
+		CloseHandle(hSnap);
+	}
+	return pid;
+}
 
 int APIENTRY WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine, int nShowCmd) {
 	AllocConsole();
 	freopen("CONOUT$", "w", stdout);
 	HANDLE hConsole = GetStdHandle(STD_OUTPUT_HANDLE);
+
+	// Execute updateoffset.cmd in order to update before run
+	SetConsoleTextAttribute(hConsole, 9);
+	printf("[MemMan] Updating required files\n");
+	SetConsoleTextAttribute(hConsole, 15);
+	system("cmd.exe /C updateoffset.cmd");
```

Comment:
You might want to check if the file exists before doing this

Reply:
Oh you're right about it I will make the change when I'll be back at home or if you want to do it yourself ...

Reply:
Fixed :
https://github.com/Skwrr/Dexterion/pull/9#pullrequestreview-2173495653